sos/report/plugins/aap_eda.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023 Rudnei Bertol Jr <rudnei@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+class AAPEDAControllerPlugin(Plugin, RedHatPlugin):
+    short_desc = 'AAP EDA Controller plugin'
+    plugin_name = 'aap_eda'
+    profiles = ('sysmgmt', 'ansible')
+    packages = ('automation-eda-controller',
+                'automation-eda-controller-server')
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/ansible-automation-platform/",
+            "/var/log/ansible-automation-platform/eda/worker.log*",
+            "/var/log/ansible-automation-platform/eda/scheduler.log*",
+            "/var/log/ansible-automation-platform/eda/gunicorn.log*",
+            "/var/log/ansible-automation-platform/eda/activation.log*",
+        ])
+
+        self.add_forbidden_path([
+            "/etc/ansible-automation-platform/eda/SECRET_KEY",
+            "/etc/ansible-automation-platform/eda/server.cert",
+            "/etc/ansible-automation-platform/eda/server.key",
+        ])
+
+        self.add_cmd_output([
+            "aap-eda-manage --version",  # eda version
+            "ls -alhR /etc/ansible-automation-platform/",
+            "ls -alhR /var/log/ansible-automation-platform/",
+        ])
+
+        self.add_cmd_output("su - eda -c 'podman ps -a'",
+                            suggest_filename="eda_podman_ps_-a")
```

Comment:
The `containers_common` plugin supports collection of this and other data from users passed to the `rootlessusers` option for the plugin.

sos has support for presets, which can be loaded based on system configuration. Would it be acceptable for you to have a new preset for AAP EDA that automatically sets the `eda` user for this option on this plugin?

Reply:
Hey @TurboTurtle ,

Sure, I can trigger the `containers_common` and set the option for `rootlessusers`. However, it is not bringing the `podman ps -a` output from the `eda` user.  Not sure if I can trigger something else, that will collect it.

Following the files and commands collected.

```
[root@aap-EDA sos]# tar -xvf /var/tmp/sosreport-aap-EDA-2024-01-18-qtjtrqg.tar.xz -C /tmp/
sosreport-aap-EDA-2024-01-18-qtjtrqg/version.txt
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_reports/
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_reports/manifest.json
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_reports/sos.html
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_reports/sos.json
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_reports/sos.txt
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_logs/
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_logs/sos.log
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_logs/ui.log
sosreport-aap-EDA-2024-01-18-qtjtrqg/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/certs.d/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/networks/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/oci/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/oci/hooks.d/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/policy.json
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf.d/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf.d/000-shortnames.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf.d/001-rhel-shortnames.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf.d/002-rhel-shortnames-overrides.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.conf.d/force-fully-qualified-images.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.d/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.d/default.yaml
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.d/registry.access.redhat.com.yaml
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/registries.d/registry.redhat.io.yaml
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/storage.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/containers/systemd/
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/subgid
sosreport-aap-EDA-2024-01-18-qtjtrqg/etc/subuid
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_commands/
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_commands/containers_common/
sosreport-aap-EDA-2024-01-18-qtjtrqg/sos_commands/containers_common/loginctl_user-status_eda
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/containers.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/mounts.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/seccomp.json
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/selinux/
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/selinux/contexts
sosreport-aap-EDA-2024-01-18-qtjtrqg/usr/share/containers/systemd/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/.config/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/.config/containers/
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/.config/containers/containers.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/.config/containers/mounts.conf
sosreport-aap-EDA-2024-01-18-qtjtrqg/var/lib/ansible-automation-platform/eda/.config/containers/podman/
```

Reply:
This relies on `machinectl` in order to do those collections. It looks like that is currently provided by the `systemd-container` package...I'm not 100% sure but I don't believe that was the case when this plugin was first written.

If you install `systemd-container`, are you getting the info you're looking for, for non-root users?

Reply:
Yes, it works. \o/
I will discuss this with my point of contact at the AAP team, as we do not install this package by default at the AAP instances.

Reply:
I created an internal Jira to discuss the benefits of using the `containers_common` plugin. It will be taken to the productization team for further discussion.
I will work to remove the `podman ps -a` from the plugin, and commit.

Reply:
Done